apluslms_roman/utils/path_mapping.py

```diff
@@ -0,0 +1,30 @@
+from pathlib import PurePosixPath
+import json
+
+
+def get_host_path(original, mapping):
+    ret = original
+    orig_path = PurePosixPath(original)
+    for k, v in mapping.items():
+        try:
+            print(k, v)
+            relative_path = orig_path.relative_to(k)
+            ret = PurePosixPath(v).joinpath(relative_path)
+            return str(ret)
+        except ValueError:
+            print("Error when composing new path!")
+            return -1
+    return str(ret)
+
+
+def get_pair_form_env(key, json_str):
+    if key == "DOCKER_DIRECTORY_MAP":
+        try:
+            ret = json.loads(json_str)
+            if isinstance(ret, dict):
+                return ret
+        except json.decoder.JSONDecodeError:
+            print("Error, check your json string")
+            return json_str
+    else:
```

Comment:
This function has 4 return states, where one is `return None` (implicit end of function). We get there when json doesn't return a dictionary (e.g. when `json_str` has value `true`). If this `else` is removed, then this function would only need 2 return statements `return ret` and `return json_str`.

Reply:
the usage of this `else` is when the key we iterating is not the key we need to read (like `DOCKER_DIRECTORY_MAP`), we will return the origin value of the string. if we don't have this else, the function may empty all value except the value of `DOCKER_DIRECTORY_MAP`

See the usage of the function:
`env = {k:  get_pair_form_env(k, v) for k, v in environ.items() if k.startswith(env_prefix)}`